test/copy.jl

```diff
@@ -82,3 +82,29 @@ let a1 = Base.svec(1, 2, 3, []), a2 = Base.svec(1, 2, 3)
     @test a3 !== b3
     @test a3[1] === a3[2]
 end
+
+# issue #16667
+let x = BigInt[1:1000;], y = deepcopy(x)
+    # Finalize the original values to make sure the deep copy is indeed
+    # independent
+    for v in x
+        finalize(v)
+    end
+    # Allocate some memory to make it more likely to trigger an error
+    # if `deepcopy` went wrong
+    x = BigInt[1:1000;]
+    @test y == x
+end
+let x = BigFloat[1:1000;], y
```

Comment:
Sorry for a possibly naive question, but then why not making sure `z` below is local too?